explorer/interpreter/interpreter.cpp

```diff
@@ -1859,6 +1859,25 @@ auto Interpreter::StepStmt() -> ErrorOr<Success> {
                  << "Cannot instantiate abstract class "
                  << dest_class->declaration().name();
         }
+
+        if (dest_class->declaration().extensibility() ==
+            ClassExtensibility::Base) {
+          for (const auto& vt : dest_class->vtable()) {
+            const auto* const fun = vt.getValue().first;
+
+            if (!fun->is_method()) {
+              continue;
+            }
+
+            if (fun->virt_override() == VirtualOverride::Abstract) {
+              return ProgramError(stmt.source_loc())
+                     << "Cannot instantiate base class `"
+                     << dest_class->declaration().name()
+                     << "`: abstract method `" << fun->name()
+                     << "` should be implemented.";
+            }
```

Comment:
This check is probably better off in TypeChecker::DeclareClassDeclaration so you don't need to setup the the vtable traversal again.